### DISCUSSION TITLE
expose frequency and voltage angle components

### DISCUSSION THREAD

**Discussion by NuDucks** [emporia-vue-local/esphome#109] [Q&A]
Hello, 

i was wondering if it's possible to expose frequency values and voltage angle components for view in home assistant? 
i see them in the code and raw quantities from this post: 
https://github.com/emporia-vue-local/esphome/discussions/102 

just not sure how to pull them out in the config.yaml file 

also, is there only one frequency quantity available or one for each phase? 

i think it would be a cool thing to be able to log. typically during brownouts or heavily loaded peak summer days you can see quite a voltage dip on the grid along with some heavy frequency deviations. If its enough (like in a brown out) to cause a locked rotor on inductive motors during startup, it can cause damage to equipment of which the power company is typically liable for. So having some evidence laying around is never a bad thing. 

anyways thanks in advance.

**Comment by flaviut** [comment]
> also, is there only one frequency quantity available or one for each phase?

Only one frequency. You can't connect one Vue to separate [grids, or you get bad data](https://github.com/emporia-vue-local/esphome/discussions/63). However, keep in mind that the way the grid works, phases are linked together such that it is impossible for different phases to have different frequencies.

There's a phase offset for each phase.

> just not sure how to pull them out in the config.yaml file

The latest version of this component has built-in support. See the "frequency" and "phase_angle" keys [in this repo's homepage](https://github.com/emporia-vue-local/esphome).

~There's *experimental* support for this if you change the source in the config to `github://SpecialMike/esphome@emporia-frequency`. See the example at https://github.com/emporia-vue-local/esphome/pull/88.~

~Please let me know if that works for you and appears accurate.~

**Reply by NuDucks** [reply]
Excellent! Thank you. Will give it a shot. Also....Yeah,your right. Brain is not 100% turned on today. Two frequencies quantities makes absolutely zero sense.

**Reply by flaviut** [reply]
The latest version of this component has built-in support! See the "frequency" and "phase_angle" keys [in this repo's homepage](https://github.com/emporia-vue-local/esphome).
